src/SpaceStation.cpp

```diff
@@ -440,8 +446,50 @@ void SpaceStation::DockingUpdate(const double timeStep)
 	for (Uint32 i=0; i<m_shipDocking.size(); i++) {
 		shipDocking_t &dt = m_shipDocking[i];
 		if (!dt.ship) continue;
-		// docked stage is m_type->NumDockingPorts() + 1 => ship docked
-		if (dt.stage > m_type->NumDockingStages()) continue;
+		if (dt.stage > m_type->NumDockingStages()) {
```

Comment:
So, let me see if I understand this correctly : we have some sort of correspondance between the ship docking stage number and the station docking stage number, up until a certain point where the station number goes higher to encode the substeps needed for the smooth land ?